### PR TITLE
Add Type Definitions for sumologic

### PR DESCRIPTION
**Pull request by banadiga**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/sumologic/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Type definitions for sumologger 1.0
+// Project: https://github.com/helix-collective/node-sumologic
+// Definitions by: banadiga <https://github.com/banadiga>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare namespace SumoLogger {}
```
Comment (RyanCavanaugh) [inline types/sumologic/index.d.ts]: Remove this

File: types/sumologic/tslint.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-unnecessary-generics": false
```
Comment (RyanCavanaugh) [inline types/sumologic/tslint.json]: This doesn't seem needed?

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

**Comment by banadiga** [comment]
Hi @RyanCavanaugh Can you please look at this PR? Thanks!

File: types/sumologic/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare class SumoLogger {
+    constructor(collectorCode: string, opts?: object);
+
+    replaceConsole(): void;
+    restoreConsole(): void;
+    augmentConsole(): void;
+    log(): void;
```
Comment (sheetalkamat) [inline types/sumologic/index.d.ts]: This seems incorrect definition.. as seems like it accepts string as per https://github.com/helix-collective/node-sumologic#usage

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Pull request closed without merging**